src/PeerPoolMgr.h

```diff
@@ -69,6 +92,9 @@ class PeerPoolMgr: public AsyncJob
     JobWait<Security::BlindPeerConnector> encryptionWait;
 
     unsigned int addrUsed; ///< counter for cycling through peer addresses
+
+public:
+    DetailedCodeContext::Pointer context;
```

Comment:
Adding CodeContext to PeerPoolMgr job is a red flag for me: We have numerous AsyncJobs that do master transaction-related work. Virtually none[^1] of them store a CodeContext pointer to report that master transaction -- after their ALE-based code context is established, those jobs _automatically_ get the right context via AsyncCalls and such. DetailedCodeContext is different from ALE-based context, but I see no reason for PeerPoolMgr job to be different with regard to its overall context _storage_ approach.

If we can apply existing ALE CodeContext handling/storage approach to PeerPoolMgr jobs without heroic efforts, let's do that instead of adding this data member.

[^1]: TunnelStateData stores CodeContext but with a TODO to remove that storage.

Reply:
Ok, let's move it from the job, but we have to keep it somewhere anyway in order to pass to PeerPoolMgr::checkpoint(). I moved it to CachePeer::standby structure, which looks logical.

Reply:
> Ok, let's move it from the job, 

Hm... Its precise location (inside the job) was not the red flag I was trying to describe. The red flag was the addition of that CodeContext data member (as such, in any class), not the addition of PeerPoolMgr::context (specifically). Sorry.

> we have to keep it somewhere anyway in order to pass to PeerPoolMgr::checkpoint()

Aha, PeerPoolMgr::Checkpoint() is the piece I was missing! We have to store this CodeContext (somewhere) because PeerPoolMgr::Checkpoint() is not called via events or async calls and, hence, cannot reuse those mechanisms for relaying context back into PeerPoolMgr job. PeerPoolMgr is probably the best location for this context storage because this context is specific to PeerPoolMgr. Please undo that move.